File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/ranges/ShortRange.java
package com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.ranges;

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.Targetable;

public class ShortRange<T extends Targetable> extends CircularRange<T> {
	
	@SafeVarargs
	public ShortRange(Class<T> targetType, TargetConstraint<T, Character> ...constraints) {
		super(targetType, constraints);
	}

	@Override
	protected double getRadius(Character character) {
		return 3.0;
	}
}

File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/ranges/ShortRange.scala
package com.anathema_roguelike
package entities.characters.perks.actions.targetingstrategies.ranges

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.Targetable
import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.constraints.TargetConstraint

import scala.reflect.runtime.universe._
import com.anathema_roguelike.entities.characters.Character

class ShortRange[T <: Targetable : TypeTag](val constraints: TargetConstraint[T, Character]*) extends CircularRange[T](constraints:_*) {
  protected def getRadius(character: Character) = 3.0
}